MAHS/Assets/SimpleEnv/Scripts/AgentScript.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using UnityEngine;
+
+using Unity.MLAgents;
+using Unity.MLAgents.Sensors;
+using Unity.MLAgents.Actuators;
+using Unity.MLAgents.Policies;
+using Debug = UnityEngine.Debug;
+
+public class AgentScript : Agent
+{
+    public enum Team
+    {
+        Hider = 0,
+        Seeker = 1
+    }
+
+    private const int NUM_TEAMS = (int) Team.Seeker;
+    
+    public TrainingAreaScript area;
+    private Rigidbody rBody;
+    private ViewField vf;
+    BehaviorParameters behaviorParameters;
+    [HideInInspector]
+    public Team team;
+    void Start()
+    {
+        rBody = GetComponent<Rigidbody>();
+        vf = GetComponent<ViewField>();
+        behaviorParameters = GetComponent<BehaviorParameters>();
+        
+        if (behaviorParameters.TeamId == (int) Team.Hider)
+        {
+            team = Team.Hider;
+        }
+        else
+        {
+            team = Team.Seeker; 
+        }
+
+        this.MaxStep = area.episodeLength;
+        
+        area.agents.Add(this);
+    }
+    
+    public override void OnEpisodeBegin()
+    {
+        this.rBody.angularVelocity = Vector3.zero;
+        this.rBody.velocity = Vector3.zero;
+        this.transform.Rotate(0,Random.value * 360 - 180f, 0);
+        
+        if (team == Team.Hider)
+        {
+            this.transform.localPosition = new Vector3(2 * (Random.value * 4.25f + 0.25f), 0.5f, 2 * (-Random.value * 4.25f - 0.25f));
```

Comment:
Let's extract all these magic constants so it's easy to change them when the time comes.